packages/react/src/components/ContextMenu/ContextMenu.tsx

```diff
@@ -0,0 +1,234 @@
+import * as React from 'react'
+import * as PropTypes from 'prop-types'
+import * as _ from 'lodash'
+import * as customPropTypes from '@stardust-ui/react-proptypes'
+
+import {
+  AutoControlledComponent,
+  RenderResultConfig,
+  applyAccessibilityKeyHandlers,
+  getOrGenerateIdFromShorthand,
+  commonPropTypes,
+} from '../../lib'
+import { ShorthandValue, Omit } from '../../types'
+
+import { Accessibility, AccessibilityAttributes } from '../../lib/accessibility/types'
+import { createShorthandFactory } from '../../lib/factories'
+import Popup, { PopupProps } from '../Popup/Popup'
+import { Menu, MenuItemProps, MenuProps, Ref } from '../..'
+import { contextMenuBehavior } from '../../lib/accessibility'
+import { focusMenuItem, focusNearest } from './focusUtils'
+import { ALIGNMENTS, POSITIONS } from '../../lib/positioner'
+
+export interface ContextMenuSlotClassNames {
+  menu: string
+}
+
+export interface ContextMenuProps
+  extends Omit<PopupProps, 'content' | 'renderContent' | 'children' | 'trapFocus'> {
+  /**
+   * Accessibility behavior if overridden by the user.
+   * @default contextMenuBehavior
+   */
+  accessibility?: Accessibility
+
+  menu?: ShorthandValue<MenuProps>
+}
+
+export interface ContextMenuState {
+  open: boolean
+  menuId: string
+  triggerId: string
+  autoFocus: boolean
+}
+
+/**
+ * A ContextMenu displays a menu connected to trigger element.
+ * @accessibility
+ */
+export default class ContextMenu extends AutoControlledComponent<
+  ContextMenuProps,
+  ContextMenuState
+> {
+  static displayName = 'ContextMenu'
+
+  static className = 'ui-contextmenu'
+
+  static create: Function
+
+  static slotClassNames: ContextMenuSlotClassNames = {
+    menu: `${ContextMenu.className}__menu`,
+  }
+
+  static propTypes = {
+    ...commonPropTypes.createCommon({
+      animated: false,
+      as: true,
+      content: false,
+    }),
+    align: PropTypes.oneOf(ALIGNMENTS),
```

Comment:
Do we want to put all Popup props here explicitly? If yes, should we have a unit test that makes sure these two are in sync?